### PR TITLE
Configuration guide - fix injecting non-existing properties

### LINKED ISSUE #2264: Configuration guide - incorrect description of injecting non-existing properties
The guide: _"If you do not provide a value for the first property, it will have a null value."_ However, injecting a non-existent property results in `DeploymentException` during app startup, i.e. something like `No config value of type [class java.lang.String] exists for: greeting.message`.

This behavior is defined by the MP Config spec.

### PR DESCRIPTION
**Pull request by mkouba**
- resolves #2264

### CODE REVIEW

File: docs/src/main/asciidoc/application-configuration-guide.adoc [hunk 1]
```diff
@@ -76,12 +76,13 @@ private String suffix;
...
+NOTE: When injecting a configured value, you can use `@Inject @ConfigProperty` or just `@ConfigProperty`.
 The `@Inject` annotation is not necessary for members annotated with `@ConfigProperty`, a behavior which differs from https://microprofile.io/project/eclipse/microprofile-config[MicroProfile Config]
 
-If you do not provide a value for the first property, it will have a `null` value.
-The second property injects the given default value if the configuration file does not provide a value.
-The third property is optional. The injected `Optional` is empty as the configuration file does not provide a value.
+If you do not provide a value for the first property, the application startup fails with `javax.enterprise.inject.spi.DeploymentException: No config value of type [class java.lang.String] exists for: greeting.message`.
+The second property injects the given default value if the configuration does not provide a value.
+The third property is optional. 
+The injected `Optional` is empty if the configuration does not provide a value.
```
Comment (gsmet) [inline docs/src/main/asciidoc/application-configuration-guide.adoc]: Minor nitpick but can we have a separate paragraph for each property (we would need an additional new line after each property)? That would be more readable.

Another option would be to use callouts.

Comment (mkouba) [reply docs/src/main/asciidoc/application-configuration-guide.adoc]: I like the callouts idea.. ;-)

**Comment by mkouba** [comment]
@gsmet changed to callouts...

**Review by gsmet — APPROVED** [review]
Perfect, thanks!

**Pull request merged**